x/farming/types/plan.go

```diff
@@ -279,6 +290,19 @@ func ValidateRatioPlans(i interface{}) error {
 	return nil
 }
 
+// PackPlan converts PlanI to Any
+func PackPlan(plan PlanI) (*codectypes.Any, error) {
+	msg, ok := plan.(proto.Message)
+	if !ok {
+		return nil, fmt.Errorf("cannot proto marshal %T", plan)
+	}
```

Comment:
IMO, we don't need this type assertion. `codectypes.NewAnyWithValue` will accept `PlanI` as `proto.Message`.